stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/attachments/content/MediaAttachmentContent.kt

```diff
@@ -0,0 +1,485 @@
+ * and the onLongItemClick handler.
+ * @param modifier The modifier used for styling.
+ * @param maximumNumberOfPreviewedItems The maximum number of thumbnails that can be displayed
+ * in a group when previewing Media attachments in the message list.
+ * @param playButton Represents the play button that is overlaid above video attachment
+ * previews.
+ */
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+public fun MediaAttachmentContent(
+    attachmentState: AttachmentState,
+    modifier: Modifier = Modifier,
+    maximumNumberOfPreviewedItems: Int = 4,
+    playButton: @Composable () -> Unit = { PlayButton() },
+) {
+    val (message, onLongItemClick, _, onMediaGalleryPreviewResult) = attachmentState
+    val gridSpacing = ChatTheme.dimens.attachmentsContentMediaGridSpacing
+
+    Row(
+        modifier
+            .clip(ChatTheme.shapes.attachment)
+            .combinedClickable(
+                indication = null,
+                interactionSource = remember { MutableInteractionSource() },
+                onClick = {},
+                onLongClick = { onLongItemClick(message) }
+            ),
+        horizontalArrangement = Arrangement.spacedBy(gridSpacing)
+    ) {
+        val attachments =
+            message.attachments.filter {
+                !it.hasLink() && it.type == AttachmentType.IMAGE || it.type == AttachmentType.VIDEO
+            }
+        val attachmentCount = attachments.size
+
+        if (attachmentCount == 1) {
+            val attachment = attachments.first()
+
+            ShowSingleMediaAttachment(
+                attachment = attachment,
+                message = message,
+                onMediaGalleryPreviewResult = onMediaGalleryPreviewResult,
+                onLongItemClick = onLongItemClick,
+                playButton = playButton
+            )
+        } else {
+            ShowMultipleMediaAttachments(
+                attachments = attachments,
+                attachmentCount = attachmentCount,
+                gridSpacing = gridSpacing,
+                maximumNumberOfPreviewedItems = maximumNumberOfPreviewedItems,
+                message = message,
+                onMediaGalleryPreviewResult = onMediaGalleryPreviewResult,
+                onLongItemClick = onLongItemClick,
+                playButton = playButton
+            )
+        }
+    }
+}
+
+/**
+ * Displays a preview of a single image or video attachment.
+ *
+ * @param attachment The attachment that is previewed.
+ * @param message The original message containing the attachment.
+ * @param onMediaGalleryPreviewResult The result of the activity used for propagating
+ * actions such as media attachment selection, deletion, etc.
+ * @param onLongItemClick Lambda that gets called when an item is long clicked.
+ * @param playButton Represents the play button that is overlaid above video attachment
+ * previews.
+ */
+@Composable
+internal fun ShowSingleMediaAttachment(
```

Comment:
naming breaks compose guidelines - it shouldn't include "Show" in it - it should be a noun, not a verb.

SingleMediaAttachment and MultipleMediaAttachments is fine